File: 8week/hellol77/two_sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        answer=[]
        for i in range(len(nums)):
            flag=0
            for j in range(i+1,len(nums)):
                if nums[i]+nums[j]==target:
                    answer.append(i)
                    answer.append(j)
                    flag=1
                    break
            if flag==1:
                break

        return answer

File: 8week/seungboshim/two_sum.cpp
#include <vector>

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int maxIdx = nums.size()-1;
        vector<pair<int,int>> v;
        for (int i=0; i<=maxIdx; i++) {
            v.push_back({nums[i], i});
        }
        sort(v.begin(), v.end());

        int left = 0;
        int right = maxIdx;
        int sum = 0;

        while (left < right) {
            sum = v[left].first + v[right].first;
            // 2 7 11 15
            if (sum == target) {
                break;
            } else if (sum > target) {
                right--;
            } else {
                left++;
            }
        }



        vector<int> ans;
        ans.push_back(v[left].second);
        ans.push_back(v[right].second);
        return ans;
    }
};